implants/imix/src/main.rs

```diff
@@ -330,39 +344,59 @@ async fn main_loop(config_path: String, run_once: bool) -> Result<()> {
             let task_response_exec_finished_at = match task_is_finished {
                 true => Some(Utc::now()),
                 false => None,
             };
-            // If the task is finished or there's new data send a task_result.
-            if task_is_finished || res.len() > 0 {
+        
+            // If the task is finished or there's new data queue a new task result.
+            if task_is_finished ||  task_channel_output.len() > 0 {
                 let task_response = SubmitTaskResultInput {
                     task_id: exec_future.1.graphql_task.id.clone(),
                     exec_started_at: exec_future.1.start_time,
                     exec_finished_at: task_response_exec_finished_at,
-                    output: res.join("\n"),
+                    output:  task_channel_output.join("\n"),
                     error: None,
                 };
-                let res = tavern_client.submit_task_result(task_response).await;
-                let _submit_task_result = match res {
-                    Ok(local_val) => local_val,
-                    Err(local_err) => if debug { println!("Failed to submit task resluts:\n{}", local_err.to_string()) },
-                };
+                let mut tmp_res_list = running_task_res_map.get(&task_id).context("Failed to get task output by ID")?.clone();
+                tmp_res_list.push(task_response);
+                running_task_res_map.insert(task_id.clone(), tmp_res_list);
             }
 
-            // Only re-insert the runnine exec futures
+            // Only re-insert the still running exec futures
             if !exec_future.1.future_join_handle.is_finished() {
-                running_exec_futures.insert(exec_future.0, exec_future.1);
+                running_exec_futures.insert(task_id, exec_future.1);
             }
         }
 
+        // Iterate over queued task results and send them back to the server
+        for (task_id, task_res) in running_task_res_map.clone().into_iter() {
+            for task_response in task_res {
+                let res = tavern_client.submit_task_result(task_response).await;
+                let _submit_task_result = match res {
+                    Ok(local_val) => {
+                        running_task_res_map.remove(&task_id);
+                        local_val
+                    },
+                    Err(local_err) => if debug { println!("Failed to submit task resluts:\n{}", local_err.to_string()) },
+                };    
+            }
+            
+        }
+
         // change the reference! This is insane but okay.
         all_exec_futures = running_exec_futures;
-        if run_once { return Ok(()); };
+        all_task_res_map = running_task_res_map.clone();
+
+        // Debug loop tracker
+        if loop_count_max.is_some() {
+            loop_count += 1;
+            if (loop_count >= loop_count_max.context("loop_count_max shouldn't be None")?) { return Ok(()); };
+        }
```

Comment:
I think this can get rewritten using if let

```
if let Some(count_max) = loop_count_max {
    loop_count += 1;
    if loop_count >= count_max {
        return Ok(())
    }
} else {
    // Handle it being None
}
```

Reply:
Oooh that is cleaner 🙌